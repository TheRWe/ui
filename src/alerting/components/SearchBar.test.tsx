// Libraries
import React from 'react'
import {screen, fireEvent, prettyDOM} from '@testing-library/react'

// Components
import SearchBar from 'src/alerting/components/SearchBar'

import {renderWithReduxAndRouter} from 'src/mockState'
import {withRouterProps} from 'mocks/dummyData'

jest.mock('src/resources/components/GetResources')

const exampleSearches = ["search 1", "example 2", "text 3"];

const setup = (override = {}) => {
  const props: Parameters<typeof SearchBar>[0] = {
    ...withRouterProps,
    dispatch: jest.fn(),
    exampleSearches,
    loadRows: jest.fn(),
    state: { searchInput: "" } as any,
    children: <></>,
    placeholder: "",
    ...override,
  }

  return renderWithReduxAndRouter(<SearchBar {...props} />)
}

describe('Alerts SearchBar', () => {
  it('show/hide autocomplete', async () => {
    const {baseElement} = setup({})

    // console.log(prettyDOM(baseElement));

    const searchInput = screen.getByTestId('check-status-input')
    fireEvent.focus(searchInput)

    // console.log(prettyDOM(baseElement));

    exampleSearches.forEach(text=>{
      const autocompleteElm = screen.getAllByText(text);
      expect(autocompleteElm.length).toBe(1);
    })

    fireEvent.blur(searchInput)

    // console.log(prettyDOM(baseElement));

    // exampleSearches.forEach(text=>{
    //   const autocompleteElm = screen.getAllByText(text);
    //   expect(autocompleteElm.length).toBe(0);
    // })

  })
})
