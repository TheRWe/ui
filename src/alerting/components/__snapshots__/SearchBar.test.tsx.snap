// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alerts SearchBar matches snapshot 1`] = `
<div>
  <div
    class="status-search-bar"
  >
    <div
      class="cf-input cf-input-sm cf-input__has-icon"
      style="width: 100%;"
    >
      <div
        class="cf-status-indicator cf-status-indicator__default cf-status-indicator__sm"
        data-testid="check-status-input--default"
      >
        <div
          class="cf-status-indicator--shadow"
        />
      </div>
      <input
        autocomplete="off"
        class="cf-input-field"
        data-testid="check-status-input"
        name=""
        placeholder=""
        spellcheck="false"
        title=""
        type="text"
        value=""
      />
      <span
        class="cf-icon search cf-input-icon"
        data-testid="icon"
      />
    </div>
  </div>
</div>
`;

exports[`Alerts SearchBar matches snapshot 2`] = `
<div>
  <div
    class="status-search-bar"
  >
    <div
      class="cf-input cf-input-sm cf-input__focused cf-input__has-icon"
      style="width: 100%;"
    >
      <div
        class="cf-status-indicator cf-status-indicator__default cf-status-indicator__sm"
        data-testid="check-status-input--default"
      >
        <div
          class="cf-status-indicator--shadow"
        />
      </div>
      <input
        autocomplete="off"
        class="cf-input-field"
        data-testid="check-status-input"
        name=""
        placeholder=""
        spellcheck="false"
        title=""
        type="text"
        value=""
      />
      <span
        class="cf-icon search cf-input-icon"
        data-testid="icon"
      />
    </div>
    <div
      class="cf-dropdown-menu status-search-bar--suggestions cf-dropdown__onyx"
      data-testid="check-status-dropdown"
      style="width: 100%;"
    >
      <div
        class="ScrollbarsCustom cf-dapper-scrollbars"
        data-testid="dapper-scrollbars"
        style="position: relative; width: 100%; height: 100%; min-width: 100%; max-width: 100%; max-height: 250px;"
      >
        <div
          class="ScrollbarsCustom-Wrapper cf-dapper-scrollbars--wrapper"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; overflow: hidden;"
        >
          <div
            class="ScrollbarsCustom-Scroller"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; overflow-y: hidden; overflow-x: hidden;"
          >
            <div
              class="ScrollbarsCustom-Content cf-dapper-scrollbars--content"
              style="box-sizing: border-box; display: table-cell; min-width: 100%;"
            >
              <div
                class="cf-dropdown-menu--contents"
                data-testid="check-status-dropdown--contents"
              >
                <div
                  class="cf-dropdown-divider"
                  data-testid="dropdown-divider"
                >
                  Examples
                </div>
                <div
                  class="cf-dropdown-item status-search-bar--suggestion cf-dropdown-item__no-wrap"
                  data-testid="dropdown-item"
                >
                  <div
                    class="cf-dropdown-item--children"
                  >
                    search 1
                  </div>
                </div>
                <div
                  class="cf-dropdown-item status-search-bar--suggestion cf-dropdown-item__no-wrap"
                  data-testid="dropdown-item"
                >
                  <div
                    class="cf-dropdown-item--children"
                  >
                    example 2
                  </div>
                </div>
                <div
                  class="cf-dropdown-item status-search-bar--suggestion cf-dropdown-item__no-wrap"
                  data-testid="dropdown-item"
                >
                  <div
                    class="cf-dropdown-item--children"
                  >
                    text 3
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
